contracts/interfaces/ICover.sol

```diff
@@ -9,27 +9,44 @@ interface ICover is IERC721 {
   /* ========== DATA STRUCTURES ========== */
 
   struct StakingPool {
-    uint160 id;
-    uint96 bookedAmount;
+    address poolAddress;
+    uint96 coverAmount;
+    uint96 premiumInNXM;
   }
 
   struct Cover {
     uint24 productId;
     uint8 payoutAsset;
-    uint8 deniedClaims;
     uint96 amount;
     uint32 start;
     uint32 period;  // seconds
+    uint96 premium;
   }
 
   struct Product {
-    uint8 payoutAsset;
+    uint16 productType;
     address productAddress;
+    uint16 capacityFactor;
+    /* supported payout assets bitmap TODO: explain */
+    uint payoutAssets;
+  }
+
+  struct ProductType {
+    string descriptionIpfsHash;
+    uint8 redeemMethod;
```

Comment:
this doesn't seem used anywhere so far

Reply:
yes it's a type i negotiated with @horodnicdragos  based on his needs in assessment